pkg/workloadapi/workload.proto

```diff
@@ -220,8 +220,11 @@ message GatewayAddress {
     NamespacedHostname hostname = 1;
     NetworkAddress address = 2;
   }
-  // port to reach the gateway at
-  uint32 port = 3;
+  // port to reach the gateway at for mTLS HBONE connections
+  uint32 hbone_mtls_port = 3;
+  // port to reach the gateway at for single tls HBONE connections
+  // used for traffic originating outside the mesh reaching a waypoint-enabled destination
+  optional uint32 hbone_single_tls_port = 4;
```

Comment:
can remove optional

Is this for ztunnel hairpin https://docs.google.com/document/d/1uM1c3zzoehiijh1ZpZuJ1-SzuVVupenv8r5yuCaFshs/edit#bookmark=id.rz8yqo8feo1n

Reply:
Yep this is for the hairpin case, but the way I was going to implement this was for each waypoint to optionally support this one way TLS flow so that users who want something similar to STRICT semantics for the waypoint can get that by not including the 15006 listener